iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java

```diff
@@ -68,13 +68,17 @@ public int getConnectedTerminalCount() {
     @Override
     public Stream<TerminalExt> getConnectedTerminalStream() {
         checkValidity();
-        return terminals.stream().map(Function.identity());
+        return terminals.stream()
+                .filter(t -> t.getConnectable() != null)
+                .map(Function.identity());
     }
 
     @Override
     public Collection<TerminalExt> getTerminals() {
         checkValidity();
-        return Collections.unmodifiableCollection(terminals);
+        return terminals.stream()
+                .filter(t -> t.getConnectable() != null)
+                .collect(Collectors.toList());
```

Comment:
Should we reuse the getConnectedTerminalStream() method here ? Or maybe you intended to skip the "Function.identity()" step ?